### PR TITLE
DM-25769: Do a health check on the server

### PR DESCRIPTION
**Pull request by SimonKrughoff**

### CODE REVIEW

**Comment by SimonKrughoff** [comment]
I tested that this by hand that this does the right thing in the case that the server returns the wrong code, but I'm not sure how to unit test it.  There's not really anything I can monkey patch to trigger that, and our test credentials don't point to an actual endpoint (I'm not sure they should).  I'm reasonably OK leaving that path untested since it's an improvement on the current situation where the error is completely unintelligible to the user.

**Comment by SimonKrughoff** [comment]
I suppose I could modify the API to allow for overriding the endpoint, but then we get back into the situation we were in before where there were several places to get credentials.  Maybe the endpoint is different enough from the username/password to justify that?

**Review by afausti — CHANGES_REQUESTED** [review]
LGTM. I have a minor comment on using f-strings.

File: lsst_efd_client/efd_helper.py [hunk 1]
```diff
@@ -49,6 +50,11 @@ def __init__(self, efd_name, db_name='efd', port='443',
         self.auth = NotebookAuth(service_endpoint=creds_service)
         host, user, password = self.auth.get_auth(efd_name)
         if client is None:
+            response = requests.get('https://'+host+'/'+'health')
```
[the remark this answers is not in the document]
Comment (SimonKrughoff) [reply lsst_efd_client/efd_helper.py:53]: Sounds good

File: lsst_efd_client/auth_helper.py [hunk 2]
```diff
@@ -1,17 +1,14 @@
 """Authentication helpers
 """
 
-import os
-import stat
-import yaml
```
[the remark this answers is not in the document]
Comment (SimonKrughoff) [reply lsst_efd_client/auth_helper.py:6]: That's exactly how I found them 👍 
I thought the github checks were running flake8, but I guess not

**Review by afausti — APPROVED** [review]

**Pull request merged**